src/receive_imf.rs

```diff
@@ -2373,12 +2377,24 @@ async fn has_verified_encryption(
                     if let Some(fp) = peerstate.gossip_key_fingerprint.clone() {
                         peerstate.set_verified(PeerstateKeyType::GossipKey, fp, verifier_addr)?;
                         peerstate.save_to_db(&context.sql).await?;
+
+                        if !is_verified {
+                            let (to_contact_id, _) = Contact::add_or_lookup(
+                                context,
+                                "",
+                                ContactAddress::new(&to_addr)?,
+                                Origin::Hidden,
+                            )
+                            .await?;
+                            ChatId::set_protection_for_contact(context, to_contact_id).await?;
```

Comment:
This line somehow makes the two jsonrpc tests time out

Reply:
It's an existing bug: https://github.com/deltachat/deltachat-core-rust/issues/4982